tests/component-tests/pager/js/PagedTableTests.js

```diff
@@ -150,62 +159,99 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
-        ];
-
         var sequence = [];
+        var linksList = pager.dataModel.map(function(data) {
+            return {text: data.animal};
+        });
 
-        var assertVisibleTips = function (message, targetIds) {
+        /* Change the page size and verify whether the tooltips and links are synced accordingly */
+        for (var pageSize = 2; pageSize <= 4; pageSize++) {
             sequence.push({
-                event: "{trackTooltips}.events.notifyFocusChange",
-                listener: function () {
-                    fluid.tests.tooltip.assertVisible(message, pager, targetIds, null, function (tooltip) {
-                        jqUnit.assertNode("The contents of the tooltip should be set", tooltipContents[0], $("b", tooltip));
-                    });
-                }
+                func: "fluid.tests.setPageSize",
+                args: [pager, pageSize]
             });
-        };
-
-        var tooltipTest = function (location) {
-            return function (idx, linkEl) {
-                var linkId = linkEl.id;
-                var link = $(linkEl);
-                sequence.push({
-                    element: link,
-                    jQueryTrigger: "focus"
-                });
-
-                if (link.hasClass(pager.pagerBar.options.styles.currentPage)) { // assumption that this will not change between now and then!
-                    assertVisibleTips("There shouldn't be any tooltips visible when the currentPage is focused", []);
-                } else {
-                    assertVisibleTips("Only the tooltip for page link " + (idx + 1) + ", in the " + location + " page bar is visible", [linkId]);
-                }
-                sequence.push({
-                    element: link,
-                    jQueryTrigger: "blur"
-                });
-                assertVisibleTips("There shouldn't be any tooltips visible when none of the pageLinks are focused", []);
-            };
-        };
+            sequence.push({
+                func: "fluid.tests.verifyLinksAndTooltips",
+                args: [pager, pageSize, linksList]
+            });
+        }
 
-        pageLinksTop.each(tooltipTest("top"));
-        pageLinksBottom.each(tooltipTest("bottom"));
         return {
             name: "Pager tooltip tests",
-            tests: {
-                name: "Tooltip visibility",
-                sequence: sequence
-            }
+            tests: [
+                {
+                    name: "Tooltip visibility and contents",
+                    sequence: sequence
+                }
+            ]
         };
     };
 
+    fluid.tests.getTooltipContents = function(listOfData, pageSize) {
+        var tooltipContents = [];
+        for (var i = 0; i < listOfData.length; i += pageSize) {
+            var tooltipContent = [];
+            var pageStartIndex = i;
+            var pageEndIndex = (i + pageSize -1) < listOfData.length ? (i + pageSize -1) : (listOfData.length - 1);
```

Comment:
It seems like you are just looking for the smaller of the two values, so you might be able to simplify this by using `Math.min`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/min